### ISSUE TITLE
Add support for JSON Patch

### ISSUE THREAD

**Issue by asbjornu** [Pomona/Pomona#96]
[RFC 6902](https://tools.ietf.org/html/rfc6902) defines `application/json-patch+json` as a more verbose, but after all a ratified internet standard for patching JSON resources. We should investigate the cost of implementing support for JSON Patch in version 2.0 since this will be a hard breaking change.

The benefit of implementing support for it is that it's a standard that probably supports what we need and perhaps in time can be removed from Pomona's source code, since it will have alternative (more complete and better tested) implementations. I also think JSON Patch is a bit more intuitive in its operations than our home brewed format.

**Comment by BeeWarloc** [comment]
I disagree about JSON patch being more intuitive, especially for simple scenarios. That's my original reason for ignoring it.

However, also supporting a ratified standard would probably be a good idea..

**Comment by asbjornu** [comment]
I think JSON Patch has less surprises, although it's more complex than our home brewed format. That's why I say "intuitive"; I don't mean to say it's simpler. It is not. But since JSON Patch has its own content type, it would be trivial to have support for both. I would prefer if Pomona defaulted to JSON Patch, though.

**Comment by asbjornu** [comment]
A good reason to implement JSON Patch would be for non-native Pomona clients to integrate with the server more easily. Seeing how there's [implementations](http://jsonpatch.com/#libraries) in [JavaScript](http://jsonpatchjs.com/), [Python](https://github.com/stefankoegl/python-json-patch), [PHP](https://github.com/mikemccabe/json-patch-php), [Ruby](https://rubygems.org/gems/json_patch), [Perl](https://github.com/zigorou/perl-json-patch), [Java](https://github.com/flipkart-incubator/zjsonpatch), [C#](https://github.com/myquay/JsonPatch), [Go](https://github.com/evanphx/json-patch), [Haskell](https://github.com/GallagherCommaJack/Haskell-JSON-Patch) and [Erlang](https://github.com/marianoguerra/json-patch.erl), plus an extensive [test suite](https://github.com/json-patch/json-patch-tests), the case in favor of adding support for `application/json-patch+json` is pretty strong, imho. 😃
